src/Points.ts

```diff
@@ -0,0 +1,146 @@
+interface ITotalOfChar {
+  letters: {
+    uppercase: number;
+    lowercase: number;
+  };
+  numbers: number;
+  symbols: number;
+}
+
+type THeightPoints = 25 | 10 | 5 | 0;
+type TMediumPoints = 20 | 10 | 5 | 0;
+type TLowPoints = 5 | 2 | 3 | 0;
+
+interface IPoints {
+  length: () => THeightPoints;
+  letters: () => TMediumPoints;
+  numbers: () => TMediumPoints;
+  characters: () => THeightPoints;
+  bonus: () => TLowPoints;
+  getAll: () => number;
+}
+
+class Points implements IPoints {
+  private passwordLength: number;
+
+  private totalOfChar: ITotalOfChar;
+
+  constructor(password: string, validSymbolsChars?: string) {
+    const symbolsChar =
+      validSymbolsChars || password.split(/[A-z]|[0-9]/g).join('');
+
+    const symbolsSearchRegex: RegExp = new RegExp(
+      `\\${symbolsChar.split('').join('|\\')}`, // used \\ to treat special regex characters (*!$+) as search instead of validation
```

Comment:
I think that if your code is not understandable you need to improve it instead of adding a comment

Reply:
PS.: you can separate in a variable

Reply:
Is very understandable, the comment refers to a javascript signaling not so commonly used

Reply:
Separate in a variable that explains why it make

Reply:
separeted

Reply:
;-; I say to separate in a variable to remove comment get the comment content and put in the variable name

Reply:
do you really think this is necessary?

Reply:
I'm not understanding you, give me an example of what you want me to do

Reply:
Only explain what you are explaining in the comment in the variable name

Reply:
like this:
```ts
const SearchNextCharacterInsteadOfUseToValidation= '\\';
```

Reply:
chaged (again)